nncf/onnx/statistics/collectors.py

```diff
@@ -30,169 +26,12 @@
-        return ONNXNNCFTensor(input_.tensor < other.tensor)
-
-    @staticmethod
-    def stack(x: Union[List[ONNXNNCFTensor], Deque[ONNXNNCFTensor]], axis: int = 0) -> ONNXNNCFTensor:
-        x = [t.tensor for t in x]
-        return ONNXNNCFTensor(np.stack(x, axis=axis))
-
-    @staticmethod
-    def unstack(x: ONNXNNCFTensor, axis: int = 0) -> List[ONNXNNCFTensor]:
-        return [ONNXNNCFTensor(np.squeeze(e, axis)) for e in np.split(x.tensor, x.tensor.shape[axis], axis=axis)]
-
-    @staticmethod
-    def squeeze(x: ONNXNNCFTensor, dim: Optional[Union[int, Tuple[int, ...]]] = None) -> ONNXNNCFTensor:
-        return ONNXNNCFTensor(np.squeeze(x.tensor, axis=dim))
-
-    @staticmethod
-    def sum(tensor: ONNXNNCFTensor) -> ONNXNNCFTensor:
-        return ONNXNNCFTensor(np.sum(tensor.tensor))
-
-    @staticmethod
-    def quantile(
-        tensor: ONNXNNCFTensor,
-        quantile: Union[float, List[float]],
-        axis: Union[int, Tuple[int, ...], List[int]],
-        keepdims: bool = False,
-    ) -> List[ONNXNNCFTensor]:
-        result = np.quantile(tensor.tensor, quantile, axis, keepdims=keepdims)
-        return [ONNXNNCFTensor(x) for x in result]
-
-    @classmethod
-    def percentile(
-        cls,
-        tensor: ONNXNNCFTensor,
-        percentile: Union[float, List[float]],
-        axis: Union[int, Tuple[int, ...], List[int]],
-        keepdims: bool = False,
-    ) -> List[TensorElementsType]:
-        quantile = np.true_divide(percentile, 100)
-        return cls.quantile(tensor, quantile=quantile, axis=axis, keepdims=keepdims)
-
-    @staticmethod
-    def mean_per_channel(x: ONNXNNCFTensor, axis: int) -> ONNXNNCFTensor:
-        if len(x.shape) < 3:
-            return ONNXNNCFTensor(np.mean(x.tensor, axis=0))
-        x = np.moveaxis(x.tensor, axis, 1)
-        t = x.reshape(x.shape[0], x.shape[1], -1)
-        return ONNXNNCFTensor(np.mean(t, axis=(0, 2)))
-
-    @staticmethod
-    def transpose(x: ONNXNNCFTensor, axes: Tuple[int, ...]) -> ONNXNNCFTensor:
-        return ONNXNNCFTensor(np.transpose(x.tensor, axes))
-
-    @staticmethod
-    def reshape(x: ONNXNNCFTensor, shape: Tuple[int, ...]) -> ONNXNNCFTensor:
-        return ONNXNNCFTensor(np.reshape(x.tensor, shape))
-
-    @staticmethod
-    def cat(x: List[ONNXNNCFTensor], axis: int) -> ONNXNNCFTensor:
-        x = [t.tensor for t in x]
-        return ONNXNNCFTensor(np.concatenate(x, axis))
-
-    @staticmethod
-    def batch_mean(x: ONNXNNCFTensor) -> ONNXNNCFTensor:
-        return ONNXNNCFTensor(np.mean(x.tensor, axis=0, keepdims=True))
-
-    @staticmethod
-    def sub(a: ONNXNNCFTensor, b: ONNXNNCFTensor) -> ONNXNNCFTensor:
-        return ONNXNNCFTensor(a.tensor - b.tensor)
-
-    @staticmethod
-    def zero_elements(x: ONNXNNCFTensor) -> ONNXNNCFTensor:
-        np_tensor = x.tensor
-        eps = np.finfo(np_tensor.dtype).eps
-        return ONNXNNCFTensor(np.abs(np_tensor) < eps)
-
-
 class ONNXBasicReducer(TensorReducerBase):
```

Comment:
Can we remove all `ONNX` and `PT` reducers and replace them by the base classes? And left reimplementation only for the `OV` as only the `OV` has inplace fns?

Reply:
Yes, it's possible but in other PR

Reply:
But this classes become redundant in this pr, why should we split the work? Why we are refactoring only the one part of the reducers and keep the other? Do we really need a commit with redundant classes? 

We could just agree to keep redundant classes as it is, don't see the point to make a separate pr